src/character.cpp

```diff
@@ -10961,6 +10962,29 @@ int Character::intimidation() const
     return ret;
 }
 
+bool Character::has_proficiency( const proficiency_id &prof ) const
+{
+    return _proficiencies.count( prof );
+}
+
+void Character::add_proficiency( const proficiency_id &prof )
+{
+    _proficiencies.insert( prof );
+}
+
+const std::vector<proficiency_id> &Character::sorted_proficiencies() const
+{
+    static std::vector<proficiency_id> sorted_profs;
```

Comment:
Please don't make this static, and return an actual values instead of a reference. Or make this cached vector a private member of the class.